src/map/battle.cpp

```diff
@@ -8271,10 +8270,22 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
 					mflag &= ~NPC_EARTHQUAKE_FLAG; // Remove before NK_SPLASHSPLIT check
 				}
 
-				if (src->type == BL_PC)
+				// TODO: This code is only accurate for pre-renewal
+				// In renewal, monsters should use NPC_EARTHQUAKE_K instead, but it's not implemented yet
+				if (sd != nullptr) {
+#ifdef RENEWAL
 					ad.damage = sstatus->str * 2 + battle_calc_weapon_attack(src, target, skill_id, skill_lv, mflag).damage;
-				else
+#else
+					ad.damage = sd->battle_status.batk + sd->battle_status.rhw.atk;
+#endif
+				}
+				else {
 					ad.damage = battle_calc_base_damage(src, sstatus, &sstatus->rhw, sc, tstatus->size, 0);
+#ifndef RENEWAL
+					if (sc != nullptr)
```

Comment:
I would move this check inside the else, as an else if, since the renewal code didn't check for the sc either,

Reply:
This won't work. You need to call battle_calc_base_damage even if sc is nullptr.

Reply:
Oh I see 👍